packages/core/core/src/TargetResolver.js

```diff
@@ -37,6 +41,228 @@ const DEFAULT_PRODUCTION_ENGINES = {
 const DEFAULT_DIST_DIRNAME = 'dist';
 const COMMON_TARGETS = ['main', 'module', 'browser', 'types'];
 
+function parseEngines(
+  targetName: ?string,
+  engines: mixed
+): Engines | typeof undefined {
+  if (engines === undefined) {
+    return;
+  } else if (engines && typeof engines === 'object') {
```

Comment:
by not differentiating between null and undefined, this can just be:

```
): ?Engines {
  if (engines == null) {
    return;
  } else if (typeof engines === 'object') {
```

Reply:
Using a schema instead now